server/unittools.h

```diff
@@ -105,14 +107,21 @@ int get_unit_vision_at(struct unit *punit, const struct tile *ptile,
 void unit_refresh_vision(struct unit *punit);
 void unit_list_refresh_vision(struct unit_list *punitlist);
 
-/// Why do we need to bounce a unit?
-enum class bounce_reason {
-  generic,        ///< We just need to do it
-  terrain_change, ///< We need to do it because of changing terrain
+struct bounce_event {
+  struct unit *bunit;
+  struct tile *to_tile;
 };
-void bounce_unit(struct unit *punit, bool verbose,
-                 bounce_reason reason = bounce_reason::generic,
-                 int max_distance = 2);
+void report_unit_bounced_to_resolve_stack_conflicts(
+    struct bounce_event bevent);
+void report_unit_disbanded_to_resolve_stack_conflicts(
```

Comment:
Are they really only used to solve stack conflicts?

Reply:
That's how the translation note is worded. If it's not, we need separate messages for different usages. (Which I believe we definitely should, but that's outside the scope of a refactor.)